api/core/model_runtime/model_providers/anthropic/llm/llm.py

```diff
@@ -538,6 +538,10 @@ def _convert_prompt_messages(self, prompt_messages: list[PromptMessage]) -> tupl
                 else:
                     raise ValueError(f"Got unknown type {message}")
 
+        # if no user message, add a system message as the user message to avoid at least one message error
+        if not prompt_message_dicts:
+            prompt_message_dicts.append({"role": "user", "content": system})
```

Comment:
This approach might lead to submitting the prompt twice, which I don't think is a very elegant solution.

Reply:
The prompt won't be submitted twice, as it only happened when prompt_message_dicts is empty, right?

Reply:
Claude has a `system` parameter in its API, which is returned in the function you modified. After your modification, it will send the system prompt twice: one in the `system` parameter, and another in `messages`(as a user prompt).

Reply:
On the other hand, the system prompt differs from the user prompt. I don't want to confuse the user by converting the system prompt into a user prompt